File: Extensions/Tessa.Extensions.Default.Client/Views/TagsInFirstColumnWorkplaceViewComponentExtension.cs
using System.Linq;
using Tessa.Platform;
using Tessa.Tags;
using Tessa.UI.Views;
using Tessa.UI.Views.Content;

namespace Tessa.Extensions.Default.Client.Views
{
    /// <summary>
    /// Расширение, которое в рабочем узле, где в метаданных для представления или пользователем в настройках таблицы задана позиция тегов "InColumn", 
    /// перемещает колонку с тегами влево.
    /// </summary>
    /// <remarks>
    /// У расширения есть конфигуратор <see cref="TagsInFirstColumnWorkplaceViewComponentExtensionConfigurator"/>.
    /// </remarks>
    public sealed class TagsInFirstColumnWorkplaceViewComponentExtension : IWorkplaceViewComponentExtension
    {
        #region IWorkplaceViewComponentExtension Implementation

        public void Clone(IWorkplaceViewComponent source, IWorkplaceViewComponent cloned, ICloneableContext context)
        {
        }

        public void Initialize(IWorkplaceViewComponent model)
        {
        }

        public void Initialized(IWorkplaceViewComponent model)
        {
            var tableView = model.Content.OfType<TableView>().FirstOrDefault();
            if (tableView?.Grid is {})
            {
                tableView.Grid.Info[TagsHelper.TagsInFirstColumnKey] = BooleanBoxes.True;
            }
        }

        #endregion
    }
}

File: WebClient SDK/src/default/views/tagsInFirstColumnWorkplaceViewComponentExtension.ts
import { extension } from '@tessa/application';
import { moveTagToFirstColumn } from 'tessa/ui/tags/tagViewsHelper';
import { IWorkplaceViewComponent } from 'tessa/ui/views';
import { TableGridViewModel } from 'tessa/ui/views/content';
import { WorkplaceViewComponentExtension } from 'tessa/ui/views/extensions';

@extension({ name: 'TagsInFirstColumnWorkplaceViewComponentExtension' })
export class TagsInFirstColumnWorkplaceViewComponentExtension extends WorkplaceViewComponentExtension {
  getExtensionName(): string {
    return 'Tessa.Extensions.Default.Client.Views.TagsInFirstColumnWorkplaceViewComponentExtension';
  }
  initialized(model: IWorkplaceViewComponent): void {
    const table = model.content.get('Table');
    if (table instanceof TableGridViewModel) {
      table.tagsInFirstColumn = true;

      if (moveTagToFirstColumn(table)) {
        table.rebuild();
      }
    }
  }
}
